.infrastructure/55_cognito.tf

```
@@ -0,0 +1,117 @@
+resource "aws_cognito_user_pool" "devportal" {
+  name                = "devportalpool"
+  deletion_protection = "ACTIVE"
+
+  user_pool_add_ons {
+    advanced_security_mode = "OFF"
+  }
+
+  username_attributes      = ["email"]
+  auto_verified_attributes = ["email"]
+
+  account_recovery_setting {
+    recovery_mechanism {
+      name     = "verified_email"
+      priority = 1
+    }
+  }
+
+  admin_create_user_config {
+    # allow user to create via signup page
+    allow_admin_create_user_only = false
+  }
+
+  password_policy {
+    minimum_length = 8
```

Just to prevent to have the following output on plan and apply:

```
  # aws_cognito_user_pool.devportal will be updated in-place
  ~ resource "aws_cognito_user_pool" "devportal" {
        id                        = "eu-south-1_rT1F5VgfO"
        name                      = "devportalpool"
        tags                      = {}
        # (10 unchanged attributes hidden)

      ~ password_policy {
          - temporary_password_validity_days = 7 -> null
            # (5 unchanged attributes hidden)
        }

        # (13 unchanged blocks hidden)
    }
```

```suggestion
    minimum_length                   = 8
    temporary_password_validity_days = 7
```